dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java

```diff
@@ -463,6 +470,30 @@ protected boolean usingProxy() {
         protected void setFixedLengthStreamingMode(int i) {
             // [CXF-6227] do not call connection.setFixedLengthStreamingMode(i)
             // to prevent https://bugs.openjdk.java.net/browse/JDK-8044726
+            
+            //Allow this mode to be set only if the Expect header has been set to "100-continue"           
+            if ((message != null) && (allowFixedLengthStreamMode == null)) {
+                // If the Expect header is set to "100-continue" then currently the JDK
+                // will not handle the returned 100 and will pass that message on the JAXRS (that is expecting
+                // the 200, not the 100).   This causes issues and potentially hangs.   To work around this
+                // issue we will allow URLConnectionHttpConduit.setFixedLengthStreamingMode() to be used.  
+                // Normally FixedLengthStreamMode is disabled due to a JDK performance issue [CXF-6227].
+                Map<String, List<String>> headers = CastUtils.cast((Map<?, ?>)message.get(Message.PROTOCOL_HEADERS));
+                List<String> expectList = headers.get("Expect");
+                
+                if (expectList != null) {
+                    Iterator<String> expects = expectList.iterator();
+                    while (expects.hasNext()) {
+                        if (expects.next().equalsIgnoreCase("100-continue")) {
+                            allowFixedLengthStreamMode = Boolean.TRUE;                            
+                        }
+                    }
+                }               
+
+            }
+            if ((allowFixedLengthStreamMode != null) && (allowFixedLengthStreamMode.booleanValue())) {
```

Comment:
All of this method (except possibly iterating over the `Expect` header values) still gets called on every request - the only difference is that once a 100-Continue header is found, then it avoids checking for the header on future requests.  Most users are probably not using `Expect: 100-Continue`, so it will slow them down on every request.  If the performance team is available to compare the results of a run with and without this change, it would be nice to see how much of an impact this is to most users.